lib/fluent/plugin/in_beats.rb

```diff
@@ -126,6 +126,7 @@ def run
           rescue => e
             log.error "unexpected error", :error => e.to_s
             log.error_backtrace
+            router.emit_error_event(tag, time, record, e)
```

Comment:
`record` may be `nil` when user doesn't specify parser. Need to check `record` / `map` value.
In addition, need option to emit error event because emit event in this point is sometimes not useful, e.g. hard to recover broken value.

Reply:
What would we be checking or comparing the record/map value to? Could you give me a quick example of how we could check this accurately?

You make a good point about the config option and not always emitting errors. I'll see if I can add that code for the config quickly.

Thanks for your feedback!